src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Deposit.tsx

```diff
@@ -278,43 +312,39 @@ export const Deposit: React.FC<DepositProps> = ({
             wallet,
           })
 
-          const fees = await adapter.getFeeData({
+          const fees = (await estimateFees({
+            accountId,
+            contractAddress: undefined,
+            assetId,
+            sendMax: false,
+            cryptoAmount: '0',
             to: customTxInput.to,
-            value: customTxInput.value,
-            chainSpecific: {
-              from: fromAccountId(accountId).account,
-              data: customTxInput.data,
-            },
-          })
+            from: fromAccountId(accountId).account,
+            memo: customTxInput.data,
+          })) as FeeDataEstimate<KnownChainIds.EthereumMainnet>
 
           const fastFeeCryptoBaseUnit = fees.fast.txFee
 
-          const fastFeeCryptoPrecision = bnOrZero(
-            bn(fastFeeCryptoBaseUnit).div(bn(10).pow(feeAsset.precision)),
-          )
+          const fastFeeCryptoPrecision = bn(fromBaseUnit(fastFeeCryptoBaseUnit, feeAsset.precision))
 
           return fastFeeCryptoPrecision.toString()
         }
 
-        // We're lying to Ts, this isn't always an UtxoBaseAdapter
-        // But typing this as any chain-adapter won't narrow down its type and we'll have errors at `chainSpecific` property
-        const adapter = chainAdapters.get(chainId) as unknown as UtxoBaseAdapter<UtxoChainId>
-        const getFeeDataInput: GetFeeDataInput<UtxoChainId> = {
+        const adapter = chainAdapters.get(chainId)!
+        const getFeeDataInput = {
           to: quote.inbound_address,
           value: amountCryptoBaseUnit,
-          // EVM chains are the only ones explicitly requiring a `from` param for the gas estimation to work
-          // UTXOs simply call /api/v1/fees (common for all accounts), and Cosmos assets fees are hardcoded
           chainSpecific: {
             pubkey: userAddress,
+            // EVM chains are the only ones explicitly requiring a `from` param for the gas estimation to work
+            // Cosmos assets fees are hardcoded, and UTXOs will never hit this method
             from: supportedEvmChainIds.includes(chainId) ? userAddress : '',
```

Comment:
even worth notating this? can just pass the address and it will be used or not depending on the chain adapter?
```suggestion
            from: userAddress,
```

Reply:
Nice catch! `from` has a diff. meaning in the context of UTXOs, though that won't matter here since we don't use this for UTXOs anymore! 5fe5cbc9a4831ca92e3fb146292337d8db58f080